### PR TITLE
Don't show submission limit warning for optional practice assignments

### LINKED ISSUE #345: Max. submissions exceeded warning in optional practice assignments
Since optional practice assignments have a maximum submission limit of 0, the submission dialog always warns the user that the maximum number of submissions has been exceeded. The warning shouldn't be shown at all for optional practice assignments

### PR DESCRIPTION
**Pull request by nikke234**
# Description of the PR

Fixes #345 

# Testing
<table>
  <tr>
    <th>unit</th>
    <th>integration</th>
    <th>manual</th>
  </tr>
  <tr>
    <td>
      <ul>
        <li>- [ ] new <b>unit</b> tests created</li>
        <li>- [x] all <b>unit</b> tests pass</li>
      </ul>
    </td>
    <td>
      <ul>
        <li>- [ ] new <b>integration</b> tests created</li>
        <li>- [x] all <b>integration</b> tests pass</li>
      </ul>
    </td>
    <td>
      <ul>
        <li>- [x] <b>manual</b> testing went well</li>
      </ul>
    </td>
  </tr>
</table>  
  
# Have you updated the [TESTING.md](/TESTING.md) or other relevant documentation?

- [ ] Yes
- [ ] Not yet. I will do it next.
- [x] Not relevant

### CODE REVIEW

File: src/main/java/fi/aalto/cs/apluscourses/ui/exercise/SubmissionDialog.java [hunk 1]
```diff
@@ -81,12 +81,12 @@ private void createUIComponents() {
...
     filenames = new JLabel(filenamesHtml.toString());
 
-    submissionCount = new JLabel(String.format("You are about to make submission %d out of %d.",
-        viewModel.getCurrentSubmissionNumber(), viewModel.getMaxNumberOfSubmissions()));
+    StringBuilder submissionCountText = new StringBuilder("You are about to make submission "
+        + viewModel.getCurrentSubmissionNumber());
+    if (viewModel.getMaxNumberOfSubmissions() != 0) {
+      submissionCountText.append(" out of " + viewModel.getMaxNumberOfSubmissions());
+    }
+    submissionCount = new JLabel(submissionCountText.toString() + ".");
```
Comment (OlliKiljunen) [inline src/main/java/fi/aalto/cs/apluscourses/ui/exercise/SubmissionDialog.java:89]: Nit: This logic could be in the view model (and unit-tested there).

Comment (OlliKiljunen) [reply src/main/java/fi/aalto/cs/apluscourses/ui/exercise/SubmissionDialog.java:89]: Also include `"."` at the end in the builder. :D

**Review by OlliKiljunen — APPROVED** [review]
Fine as it is but please also consider my comments.

File: src/main/java/fi/aalto/cs/apluscourses/ui/exercise/SubmissionDialog.java (hunk 1 above)
Comment (nikke234) [reply src/main/java/fi/aalto/cs/apluscourses/ui/exercise/SubmissionDialog.java:89]: You are correct on both points :smile: will do

**Review by superseacat — APPROVED** [review]
* looks fine
* works fine

:+1:

**Pull request merged**